### PR TITLE
extended: test for iterative deployments

### PR DESCRIPTION
**Pull request by kargakis**
Closes https://github.com/openshift/origin/issues/9652

### CODE REVIEW

**Comment by kargakis** [comment]
[testextended][extended:core(iteratively)]

**Comment by kargakis** [comment]
@mfojtik PTAL

File: test/extended/deployments/deployments.go [hunk 1]
```diff
@@ -130,6 +130,55 @@ var _ = g.Describe("deploymentconfigs", func() {
...
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			condition := func() (bool, error) {
+				dc, rcs, pods, err := deploymentInfo(oc, name)
+				if err != nil {
+					return false, nil
+				}
+
+				// Check that the deployment config has latestVersion=2
+				g.By(fmt.Sprintf("by checking that the deployment config has the correct version"))
```
Comment (mfojtik) [inline test/extended/deployments/deployments.go]: no need for Sprintf

File: test/extended/deployments/deployments.go [hunk 2]
```diff
@@ -130,6 +130,55 @@ var _ = g.Describe("deploymentconfigs", func() {
...
+						secondDeploymentExists = true
+						break
+					}
+				}
+
+				// Check that the first deployer is deleted and the second deployer exists.
+				g.By(fmt.Sprintf("by checking that the first deployer was deleted and the second deployer exists"))
+				deploymentNamesToDeployers, err := deploymentPods(pods)
+				if err != nil {
+					return false, nil
```
Comment (mfojtik) [inline test/extended/deployments/deployments.go]: should return err as well?

**Comment by mfojtik** [comment]
small nits, LGTM

File: test/extended/deployments/deployments.go (hunk 2 above)
Comment (kargakis) [inline test/extended/deployments/deployments.go]: No, we shouldn't error out from here at all. We rely on the timeout for erroring out which is pretty small anyway.

**Comment by kargakis** [comment]
images duckduckgo com
[merge]

**Pull request merged**